Intelipro/reja
# Feat: Implement Customer Life Time Values and RFM

#### What does this PR do?
Implement Customer segment Life Time Values and RFM

#### Description of Task to be completed?

- [x]  Add segment buttons (all, low, mid, high)
- [x]  Create Lifetime value cards for each segment (all, low, mid, high)
- [x]  Add Period dropdown to filter the Lifetime values
- [x]  Add a table to display customers RFM values with search and pagination functionality

#### How should this be manually tested?
- Clone this repo `git clone  [the-repo-link]`
- Checkout to this branch `ft-customer-segment`
- Install all dependencies `npm install`
- Start the server `npm start`
- To access the page go to `http://localhost:3000` on your browser 
- Log in to access the application
- Navigate to the customer page under the analytics sidebar item. Below the graph, you'll see the segment section
- Click the various value level(low, mid, high) to see the specif Lifetime value and RFM
- Use the period dropdown to see the different Lifetime values

#### Any background context you want to provide?
  RFM stands for three dimensions:
- Recency – How recently did the customer purchase?
- Frequency – How often do they purchase?
- Monetary Value – How much do they spend?

Currently, the data plotted are for a specific client (client with Id 10).

#### What are the relevant pivotal tracker stories?

#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/43441107/75422775-5d8c0500-594e-11ea-9576-8331c631bd79.png)
![image](https://user-images.githubusercontent.com/43441107/75422792-654ba980-594e-11ea-8c84-a652534bed1e.png)

#### Questions:
